core/proto/core.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto2";
+
+package core;
+
+option go_package = "proto";
+
+message Challenge {
+	optional int64 id = 1;
+	optional string type = 2;
+	optional string status = 6;
+	optional string uri = 9;
+	optional string token = 3;
+	optional string accountKey = 4; // jose.JsonWebKey.MarshalJSON()
+	optional string keyAuthorization = 5;
+	repeated ValidationRecord validationrecords = 10;
+	optional ProblemDetails error = 7;
+}
+
+message ValidationRecord {
+        optional string hostname = 1;
+        optional string port = 2;
```

Comment:
Because core.ValidationRecord.Port is a string, because the stdlib takes strings for ports.